### PR TITLE
Merged with goat_launch_laptop

### PR DESCRIPTION
**Pull request by mattsara24**
* Added nuc_competition_combined launch files
* Added demo_2d and demo_3d config files
* remove old unused files/launch scripts
* add GPS launch script
* set nodes to respawn instead of killing the entire launch process

### CODE REVIEW

File: goat_launch/config/cartographer_no_odom.lua [hunk 1]
```diff
@@ -0,0 +1,52 @@
+-- Copyright 2016 The Cartographer Authors
```
Comment (Gregory-Meyer) [inline goat_launch/config/cartographer_no_odom.lua]: delete this file, cartographer without odom is awful

File: goat_launch/launch/goat_odom.launch [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<launch>
+  <node pkg="robot_localization" type="ekf_localization_node" name="ekf_se" clear_params="true">
+    <rosparam command="load" file="$(find goat_launch)/params/goat_odom.yaml" />
+
+    <remap from="cmd_vel" to="cmd_vel_odom"/>
+
+    <!--  Placeholder for output topic remapping
```
Comment (Gregory-Meyer) [inline goat_launch/launch/goat_odom.launch]: delete this comment

File: goat_launch/launch/goat_odom.launch [hunk 3]
```diff
@@ -0,0 +1,13 @@
+<launch>
+  <node pkg="robot_localization" type="ekf_localization_node" name="ekf_se" clear_params="true">
```
Comment (Gregory-Meyer) [inline goat_launch/launch/goat_odom.launch]: rename this node to ekf_localization_node to follow convention in the rest of the repo

File: goat_launch/launch/nuc_competition_combined.launch [hunk 4]
```diff
@@ -0,0 +1,13 @@
+<launch>
```
Comment (Gregory-Meyer) [inline goat_launch/launch/nuc_competition_combined.launch]: GPS?

File: goat_launch/launch/nuc_competition_combined.launch [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+    <include file="$(find goat_launch)/launch/state.launch"/>
+    <include file="$(find goat_launch)/launch/imu.launch"/>
+    <include file="$(find goat_launch)/launch/cartographer.launch"/>
+    <include file="$(find velodyne_pointcloud)/launch/VLP16_points.launch"/>
+
+    <include file="$(find goat_launch)/launch/encoders.launch"/>
+    <include file="$(find goat_launch)/launch/goat_odom.launch"/>
+
+    <node name="cartographer_occupancy_grid_node" pkg="cartographer_ros"
```
Comment (Gregory-Meyer) [inline goat_launch/launch/nuc_competition_combined.launch]: not necessary anymore, remove the occupancy grid node

File: goat_launch/launch/nuc_competition_combined_visualize.launch [hunk 6]
```diff
@@ -0,0 +1,20 @@
+<launch>
+    <include file="$(find goat_launch)/launch/state.launch"/>
+    <include file="$(find goat_launch)/launch/imu.launch"/>
+    <include file="$(find goat_launch)/launch/cartographer.launch"/>
+    <include file="$(find velodyne_pointcloud)/launch/VLP16_points.launch"/>
+
+    <include file="$(find goat_launch)/launch/encoders.launch"/>
+    <include file="$(find goat_launch)/launch/goat_odom.launch"/>
+
+    <node name="cartographer_occupancy_grid_node" pkg="cartographer_ros"
```
Comment (Gregory-Meyer) [inline goat_launch/launch/nuc_competition_combined_visualize.launch]: remove this

File: goat_launch/launch/nuc_competition_combined_visualize.launch [hunk 7]
```diff
@@ -0,0 +1,20 @@
+<launch>
```
Comment (Gregory-Meyer) [inline goat_launch/launch/nuc_competition_combined_visualize.launch]: why not just include the nuc_competition_combined launchfile? duplicated code

File: goat_launch/launch/nuc_competition_combined_visualize.launch [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+    <include file="$(find goat_launch)/launch/encoders.launch"/>
+    <include file="$(find goat_launch)/launch/goat_odom.launch"/>
+
+    <node name="cartographer_occupancy_grid_node" pkg="cartographer_ros"
+        type="cartographer_occupancy_grid_node" args="-resolution 0.05" />
+
+    <node name="rviz" pkg="rviz" type="rviz" required="true"
+        args="-d $(find goat_launch)/config/demo_3d.rviz" />
+  
+    <node name="costmap_2d_markers" pkg="costmap_2d" type="costmap_2d_markers">
```
Comment (Gregory-Meyer) [inline goat_launch/launch/nuc_competition_combined_visualize.launch]: we don't use costmap_2d anymore, remove this

File: goat_launch/params/goat_odom.yaml [hunk 9]
```diff
@@ -0,0 +1,171 @@
...
+# unspecified.
+transform_time_offset: 0.0
+
+# Use this parameter to provide specify how long the tf listener should wait for a transform to become available. 
+# Defaults to 0.0 if unspecified.
+transform_timeout: 0.0
+
+# If you're having trouble, try setting this to true, and then echo the /diagnostics_agg topic to see if the node is
+# unhappy with any settings or data.
+print_diagnostics: true
```
Comment (Gregory-Meyer) [inline goat_launch/params/goat_odom.yaml]: maybe this should be a parameter in the launchfile instead? so we can switch it on and off without messing with YAML files

File: goat_launch/params/goat_odom.yaml [hunk 10]
```diff
@@ -0,0 +1,171 @@
...
+
+# Debug settings. Not for the faint of heart. Outputs a ludicrous amount of information to the file specified by
+# debug_out_file. I hope you like matrices! Please note that setting this to true will have strongly deleterious
+# effects on the performance of the node. Defaults to false if unspecified.
+debug: false
+
+# Defaults to "robot_localization_debug.txt" if unspecified. Please specify the full path.
```
Comment (Gregory-Meyer) [inline goat_launch/params/goat_odom.yaml]: this isn't used so remove it

File: goat_launch/params/goat_odom.yaml [hunk 11]
```diff
@@ -0,0 +1,171 @@
...
+twist0_queue_size: 3
+twist0_rejection_threshold: 2
+twist0_nodelay: false
+
+imu0: imu/data_filtered
+imu0_config: [false, false, false,
+              true,  true,  true,
+              false, false, false,
+              true,  true,  true,
+              true,  true,  true]
```
Comment (Gregory-Meyer) [inline goat_launch/params/goat_odom.yaml]: AHH do not use IMU acceleration it messes everything up

File: goat_launch/params/goat_odom.yaml [hunk 12]
```diff
@@ -0,0 +1,171 @@
...
+twist0_config: [false, false, false,
+                false, false, false,
+                true,  true,  true,
+                false, false, false,
+                false, false, false]
+twist0_queue_size: 3
+twist0_rejection_threshold: 2
+twist0_nodelay: false
+
+imu0: imu/data_filtered
```
Comment (Gregory-Meyer) [inline goat_launch/params/goat_odom.yaml]: use imu/data IIRC

File: goat_launch/params/goat_odom.yaml [hunk 13]
```diff
@@ -0,0 +1,171 @@
...
+# from landmark observations) then:
+#     3a. Set your "world_frame" to your map_frame value
+#     3b. MAKE SURE something else is generating the odom->base_link transform. Note that this can even be another state
+#         estimation node from robot_localization! However, that instance should *not* fuse the global data.
+map_frame: map              # Defaults to "map" if unspecified
+odom_frame: odom            # Defaults to "odom" if unspecified
+base_link_frame: base_link  # Defaults to "base_link" if unspecified
+world_frame: odom           # Defaults to the value of odom_frame if unspecified
+
+twist0: encoders/twist
```
Comment (Gregory-Meyer) [inline goat_launch/params/goat_odom.yaml]: shouldn't this be encoders/odom?

File: goat_launch/params/goat_odom.yaml [hunk 14]
```diff
@@ -0,0 +1,171 @@
...
+
+imu0: imu/data_filtered
+imu0_config: [false, false, false,
+              true,  true,  true,
+              false, false, false,
+              true,  true,  true,
+              true,  true,  true]
+imu0_nodelay: false
+imu0_differential: false
+imu0_relative: true
```
Comment (Gregory-Meyer) [inline goat_launch/params/goat_odom.yaml]: no

File: goat_launch/params/goat_odom.yaml [hunk 15]
```diff
@@ -0,0 +1,171 @@
...
+
+# [ADVANCED] The process noise covariance matrix can be difficult to tune, and can vary for each application, so it is
+# exposed as a configuration parameter. This matrix represents the noise we add to the total error after each
+# prediction step. The better the omnidirectional motion model matches your system, the smaller these values can be.
+# However, if users find that a given variable is slow to converge, one approach is to increase the
+# process_noise_covariance diagonal value for the variable in question, which will cause the filter's predicted error
+# to be larger, which will cause the filter to trust the incoming measurement more during correction. The values are
+# ordered as x, y, z, roll, pitch, yaw, vx, vy, vz, vroll, vpitch, vyaw, ax, ay, az. Defaults to the matrix below if
+# unspecified.
+process_noise_covariance: [0.05, 0,    0,    0,    0,    0,    0,     0,     0,    0,    0,    0,    0,    0,    0,
```
Comment (Gregory-Meyer) [inline goat_launch/params/goat_odom.yaml]: oh geez

File: goat_launch/params/goat_odom.yaml [hunk 16]
```diff
@@ -0,0 +1,171 @@
...
+                           0,    0,    0,    0,    0,    0,    0,     0,     0,    0,    0,    0,    0.01, 0,    0,
+                           0,    0,    0,    0,    0,    0,    0,     0,     0,    0,    0,    0,    0,    0.01, 0,
+                           0,    0,    0,    0,    0,    0,    0,     0,     0,    0,    0,    0,    0,    0,    0.015]
+
+# [ADVANCED] This represents the initial value for the state estimate error covariance matrix. Setting a diagonal
+# value (variance) to a large value will result in rapid convergence for initial measurements of the variable in
+# question. Users should take care not to use large values for variables that will not be measured directly. The values
+# are ordered as x, y, z, roll, pitch, yaw, vx, vy, vz, vroll, vpitch, vyaw, ax, ay, az. Defaults to the matrix below
+#if unspecified.
+initial_estimate_covariance: [1e-9, 0,    0,    0,    0,    0,    0,    0,    0,    0,     0,     0,     0,    0,    0,
```
Comment (Gregory-Meyer) [inline goat_launch/params/goat_odom.yaml]: can we delete this? we don't know what to do with it honestly

File: goat_launch/params/goat_odom.yaml (hunk 15 above)
Comment (Gregory-Meyer) [reply goat_launch/params/goat_odom.yaml]: delete this nephew

**Comment by samjhall** [comment]
lets get this bread

**Pull request merged**